presto-main/src/main/java/com/facebook/presto/sql/planner/TimestampRuleChange.java

```diff
@@ -0,0 +1,144 @@
+                filter.getSourceLocation(),
+                filter.getId(),
+                filter.getSource(),
+                rewritten);
+        return Result.ofPlanNode(newFilter);
+    }
+
+    private RowExpression rewritePredicate(RowExpression expression)
+    {
+        // If not a function call, do nothing
+        if (!(expression instanceof CallExpression)) {
+            return expression;
+        }
+        CallExpression call = (CallExpression) expression;
+
+        // If the displayName is '=' and there are exactly two arguments, check for date(...) usage
+        if ("=".equalsIgnoreCase(call.getDisplayName()) && call.getArguments().size() == 2) {
+            RowExpression left = call.getArguments().get(0);
+            RowExpression right = call.getArguments().get(1);
+
+            Optional<RowExpression> maybeCol = extractCastToDate(left);
+            Optional<LocalDate> maybeDate = extractDateLiteral(right);
+
+            if (!maybeCol.isPresent() || !maybeDate.isPresent()) {
+                // try flipping
+                maybeCol = extractCastToDate(right);
+                maybeDate = extractDateLiteral(left);
+            }
+
+            if (maybeCol.isPresent() && maybeDate.isPresent()) {
+                // We'll rewrite to: col >= TIMESTAMP 'yyyy-mm-dd 00:00:00.000'
+                LocalDate dateVal = maybeDate.get();
+                String start = dateVal.toString() + " 00:00:00.000";
+
+                return comparisonExpression(
+                        functionResolution,
+                        GREATER_THAN_OR_EQUAL,
+                        maybeCol.get(),
+                        constant(start, TIMESTAMP));
+            }
+        }
+
+        // If not recognized, leave it alone
+        return expression;
+    }
+
+    private Optional<RowExpression> extractCastToDate(RowExpression expr)
+    {
+        if (!(expr instanceof CallExpression)) {
+            return Optional.empty();
+        }
+        CallExpression call = (CallExpression) expr;
+        // Check if return type is date
+        if (!call.getType().toString().equalsIgnoreCase("date")) {
+            return Optional.empty();
+        }
+        // Check if name includes "cast"
+        if (!call.getDisplayName().toLowerCase().contains("cast")) {
+            return Optional.empty();
+        }
+        if (call.getArguments().isEmpty()) {
+            return Optional.empty();
+        }
+        // Child should be the underlying timestamp expression
+        return Optional.of(call.getArguments().get(0));
+    }
+
+    private Optional<LocalDate> extractDateLiteral(RowExpression expr)
+    {
+        if (!(expr instanceof ConstantExpression)) {
+            return Optional.empty();
+        }
+        ConstantExpression c = (ConstantExpression) expr;
+        if (c.getValue() == null) {
+            return Optional.empty();
+        }
+        try {
+            return Optional.of(LocalDate.parse(c.getValue().toString()));
```

Comment:
Rather than parse and turn this into `LocalDate`, let's just pull the Block value instead. We can operate on the block directly instead